lib/vanagon/platform.rb

```diff
@@ -117,6 +117,20 @@ def [](key)
       end
     end
 
+    # Get the output dir for packages. If the output_dir was defined already (by
+    # the platform config) then don't change it.
+    #
+    # @param target_repo [String] optional repo target for built packages defined
+    #   at the project level
+    # @return [String] relative path to where packages should be output to
+    def output_dir(target_repo = "")
+      if @output_dir
+        @output_dir
+      else
+        File.join(@os_name, @os_version, target_repo, @architecture)
+      end
```

Comment:
This could be more concisely represented like this:

```
@output_dir ||= File.join(@os_name, @os_version, target_repo, @architecture)
```

That will both use the existing value if it's defined, and populate the instance variable with the computed value if it's not -- then future calls to `#output_dir` will return the populated instance variable instead of recomputing every run-through. It's not functionally identical to the existing code (which will **always** return the computed value if `@output_dir` isn't truthy) but it's probably a preferred way to handle this value since we're pretty lazy about using the accessor method instead directly calling the instance variable.